lib/ddtrace/configuration/validation_proxy.rb

```diff
@@ -0,0 +1,101 @@
+require 'forwardable'
+
+module Datadog
+  module Configuration
+    # Forwards configuration settings that are permitted,
+    # but raises errors for access to anything else.
+    class ValidationProxy
+      extend Forwardable
+
+      FORWARDED_METHODS = [
+        :reset!,
+        :to_h
```

Comment:
It's not very big, but exposing `to_h` does leak the entire contents of the settings object:

```ruby
[11] pry(main)> Datadog::Tracing.configuration.to_h[:profiling]
=> #<#<Class:0x00007fd17b84b350>:0x00007fd17baffa60
 @options=
  {:enabled=>
    #<Datadog::Configuration::Option:0x00007fd17baff8a8
     @context=#<#<Class:0x00007fd17b84b350>:0x00007fd17baffa60 ...>,
     @definition=
      #<Datadog::Configuration::OptionDefinition:0x00007fd17b84aef0
       @default=#<Proc:0x00007fd17b84af90 /Users/ivo.anjo/datadog/dd-trace-rb/lib/ddtrace/configuration/settings.rb:247>,
       @delegate_to=nil,
       @depends_on=[],
       @lazy=true,
       @name=:enabled,
       @on_set=nil,
       @resetter=nil,
       @setter=#<Proc:0x00007fd17c9727a0 /Users/ivo.anjo/datadog/dd-trace-rb/lib/ddtrace/configuration/option_definition.rb:8 (lambda)>>,
     @is_set=true,
     @value=false>}>
```

So I would suggest not forwarding it.

Reply:
Was adding all public methods that seemed worthy, but you might be right. Just have to make sure it's not a problem for the environment logger.